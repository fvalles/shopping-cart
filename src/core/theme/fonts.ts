export const Fonts = {
  light: 'Avenir Light',
  medium: 'Avenir Medium',
  regular: 'Avenir Regular',
};

export type FontsType = typeof Fonts;
