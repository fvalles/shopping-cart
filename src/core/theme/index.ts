export { Colors } from './colors';
export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
