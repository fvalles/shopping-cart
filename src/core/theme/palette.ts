export const palette = {
  // color names are based on https://colornamer.netlify.app/
  black: '#000000',
  blackMarket: '#212240',
  blackMarketLight: '#21224029',
  bleachedSilk: '#F3F3F3',
  faintFern: '#DBDBE0',
  mischka: '#A6A7B3',
  sharpGrey: '#CACAD1',
  stormGrey: '#717285',
  venetianNights: '#7350FF',
  white: '#FFFFFF',
};
