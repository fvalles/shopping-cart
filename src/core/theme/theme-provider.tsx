import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Colors, Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Colors,
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
