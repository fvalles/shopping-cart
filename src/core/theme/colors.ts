import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  primary: palette.venetianNights,
  productImageBorder: palette.sharpGrey,
  productPrice: palette.black,
  productQuantity: palette.faintFern,
  productTitle: palette.mischka,
  summaryBackground: palette.bleachedSilk,
  summaryHeading: palette.stormGrey,
  summaryText: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
