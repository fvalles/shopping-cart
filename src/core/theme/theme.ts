import { DefaultTheme } from 'styled-components';
import { Colors } from './colors';

const theme: DefaultTheme = {
  Colors,
};

export { theme };
