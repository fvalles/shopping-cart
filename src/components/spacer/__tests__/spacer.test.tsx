import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import { Spacer } from '..';
import 'jest-styled-components';

describe('spacer component', () => {
  it('should render with default height', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Spacer />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '16px');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom height of 32px', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Spacer size={32} />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '32px');
    expect(result).toMatchSnapshot();
  });
});
