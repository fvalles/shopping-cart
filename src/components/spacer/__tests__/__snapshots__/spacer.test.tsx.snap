// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacer component should render with custom height of 32px 1`] = `
.c0 {
  height: 32px;
}

<div
  className="c0"
  size={32}
/>
`;

exports[`spacer component should render with default height 1`] = `
.c0 {
  height: 16px;
}

<div
  className="c0"
/>
`;
